src/ocean/text/util/StringC.d

```diff
@@ -89,7 +89,9 @@ class StringC
     {
         if (str.length && !!str[$ - 1])
         {
+            enableStomping(str);
             str ~= StringC.Term;
```

Comment:
This is wrong place to call `enableStomping` - it has to be done right after `str.length` is changed, not upon appending. The way you do it now simply hides missing `enableStomping` in some other place.